sdl-core/src/main/scala/io/smartdatalake/workflow/action/generic/transformer/SQLDfsTransformer.scala

```diff
@@ -22,47 +22,72 @@ package io.smartdatalake.workflow.action.generic.transformer
  */
-case class SQLDfsTransformer(override val name: String = "sqlTransform", override val description: Option[String] = None, code: Map[DataObjectId,String], options: Map[String, String] = Map(), runtimeOptions: Map[String, String] = Map()) extends OptionsGenericDfsTransformer {
+case class SQLDfsTransformer(override val name: String = "sqlTransform", override val description: Option[String] = None, code: Map[String,String], options: Map[String, String] = Map(), runtimeOptions: Map[String, String] = Map())
+  extends OptionsGenericDfsTransformer with SmartDataLakeLogger {
   override def transformWithOptions(actionId: ActionId, partitionValues: Seq[PartitionValues], dfs: Map[String,GenericDataFrame], options: Map[String, String])(implicit context: ActionPipelineContext): Map[String,GenericDataFrame] = {
     val functions = DataFrameSubFeed.getFunctions(dfs.values.head.subFeedType)
-    // register all input DataObjects as temporary table
-    dfs.foreach {
-      case (dataObjectId,df) =>
-        val objectId =  ActionHelper.replaceSpecialCharactersWithUnderscore(dataObjectId)
-        // Using createTempView does not work because the same data object might be created more than once
-        df.createOrReplaceTempView(objectId)
+    // register all inputs as temporary table
+    val inputViewNameOptions = dfs.map {
+      case (inputName,df) =>
+        val inputViewName =  ActionHelper.createTemporaryViewName(inputName)
+        // Using createTempView does not work because the same temporary view is created more than once (Init & Exec phase...)
+        df.createOrReplaceTempView(inputViewName)
+        (s"${INPUT_VIEW_NAME}_$inputName" -> inputViewName)
     }
-    // execute all queries and return them under corresponding dataObjectId
+    // get an output DataObject from the Action to use to creating DataFrame from SQL. Note that this DataObject is only used to get connection information.
+    val outputDataObjectId = context.instanceRegistry.get[Action](actionId).outputs.head.id
+    // execute all queries and return them under corresponding output name
     code.map {
-      case (dataObjectId,sql) =>
+      case (outputName,sql) =>
         val df = try {
-          val preparedSql = SparkExpressionUtil.substituteOptions(dataObjectId, Some(s"transformers.$name.code"), sql, options)
-          functions.sql(preparedSql,dataObjectId)
+          var preparedSql = SparkExpressionUtil.substituteOptions(actionId, Some(s"transformers.$name.code"), sql, options ++ inputViewNameOptions)
+          // for backward compatibility the temp view name from versions <= 2.2.x is replaced with the new temp view name including a postfix.
+          if (Environment.replaceSqlTransformersOldTempViewName) {
+            inputViewNameOptions.values.foreach(inputViewName => preparedSql = ActionHelper.replaceLegacyViewName(preparedSql, inputViewName))
```

Comment:
use a different name than preparedSql. It's not the same than the one above, right?

Reply:
Yes its the same :-) But i could use foldLeft instead...